learn/data_science/pandas.md

```diff
@@ -0,0 +1,375 @@
+  
+## What is `pandas` in Python  
+
+Pandas includes several key features that make it incredibly powerful for data analysis:
+
+1. Dataframe object for data manipulation with integrated indexing.
+2. Tools for reading and writing data between in-memory data structures and different file formats.
+3. Data alignment and integrated handling of missing data.
+4. Reshaping and pivoting of datasets.
+5. Label-based slicing, indexing, and subsetting of large datasets.
+6. Data structure column insertion and deletion.
+7. Group by engine allowing split-apply-combine operations on datasets.
+8. High performance merging and joining of datasets.
+
+### `read_csv()` and `to_csv()` Methods 
+
+One of the reasons why the [pandas](https://pypi.org/project/pandas/) library is highly regarded in the data science community is its ability to easily read and write data. This includes working with CSV files, a common data format in data analysis. Here are simple examples illustrating how to read data from a CSV file into a `pandas` DataFrame, and how to write this data back out to a CSV file.
+
+> Note: It's important to `import` the `pandas` library before executing these code snippets. This is typically done with the line `import pandas as pd`.
+
+### Example: Reading data from a CSV file
+
+```python
+import pandas as pd
+
+# Load data from a CSV file into a DataFrame
+data = pd.read_csv('example.csv')
+print(data.head())
+```
+
+### Example: Writing data to a CSV file
+
+```python
+# Assuming 'data' is a DataFrame that you have previously created and manipulated
+data.to_csv('modified_example.csv', index=False)
+```
+
+### Working with DataFrames
+
+At the core of the `pandas` library is the DataFrame object, which can be thought of as a relational data table, with rows and columns. Here's how you can create a DataFrame from scratch and inspect its contents.
+
+### Example: Сreate DataFrame()`
+
+```python
+# Creating a DataFrame from a dictionary
+df = pd.DataFrame({
+    'A': [1, 2, 3],
+    'B': ['a', 'b', 'c']
+})
+print(df)
+```
+
+Output:
+
+```python
+
+   A  B
+0  1  a
+1  2  b
+2  3  c
+```
+  
+## Install `pandas`  
+
+To get started, you need to install Pandas. This can be done using either pip or conda, depending on your Python environment.
+
+### Using pip
+
+If you're using pip, open your terminal or command prompt and execute the following command:
+
+```python
+pip install pandas
+```
+
+### Using conda
+
+For those who prefer Conda, the command differs slightly:
+
+```python
+conda install pandas
+```
+
+> Note: Ensure your internet connection is active as these commands download `pandas` from the respective repositories.
+
+### Step 2 Import Pandas
```

Comment:
If we write Step 2, the Step 1 must exists as well :) But I guess it's better to remove the Step 2 text here.